hepcrawl/spiders/desy_spider.py

```diff
@@ -78,27 +80,81 @@ class DesySpider(StatefulSpider):
     def __init__(
         self,
         source_folder=None,
-        ftp_folder='/DESY',
-        ftp_host=None,
-        ftp_netrc=None,
         destination_folder='/tmp/DESY',
+        s3_input_bucket='inspire-publishers-desy-incoming',
+        s3_output_bucket='inspire-publishers-desy-processed',
+        s3_server="https://s3.cern.ch",
+        s3_key=None,
+        s3_secret=None,
         *args,
         **kwargs
     ):
         super(DesySpider, self).__init__(*args, **kwargs)
-        self.ftp_folder = ftp_folder
-        self.ftp_host = ftp_host
-        self.ftp_netrc = ftp_netrc
+
         self.source_folder = source_folder
         self.destination_folder = destination_folder
-        self.ftp_enabled = False if self.source_folder else True
 
-        if self.ftp_enabled and not self.ftp_host:
-            raise Exception('You need to pass source_folder or ftp_host.')
+        self.s3_server = s3_server
+        self.s3_key = s3_key
+        self.s3_secret = s3_secret
+        self.s3_input_bucket = s3_input_bucket
+        self.s3_output_bucket = s3_output_bucket
+
+        self.s3_enabled = False if self.source_folder else True
+
+        if self.s3_enabled:
+            if not (self.s3_server and self.s3_key and self.s3_secret and self.s3_input_bucket and self.s3_output_bucket):
+                raise Exception("Missing s3 connection parameters")
+            else:
+                self.s3_connections = self.connect_to_s3()
+                self.s3_resource = self.get_s3_resource()
 
         if not os.path.exists(self.destination_folder):
             os.makedirs(self.destination_folder)
 
+    def get_s3_resource(self):
+        if not self.s3_enabled:
+            raise Exception("S3 is not enabled.")
+        for key, value in self.s3_connections.items():
+            if key == "s3" and hasattr(value.meta, "client"):
+               return value
+        return None
+
+    def connect_to_s3(self):
+        s3_session_params = {
+            "aws_access_key_id": self.s3_key,
+            "aws_secret_access_key": self.s3_secret
+        }
+        session = boto3.session.Session(**s3_session_params)
+        service = "s3"
+        try:
+            connections = {}
+            kwargs = s3_session_params.copy()
+            kwargs["endpoint_url"] = self.s3_server
+            # Create resource or client
+            if service in session.get_available_resources():
+                connections.update({service: session.resource(service, **kwargs)})
+            else:
+                connections.update({service: session.client(service, **kwargs)})
+        except UnknownServiceError as exc:
+            self.logger.error("Cannot connect to s3.", exc)
+            raise
+        return connections
+    def get_s3_url_for_key(self, key, bucket=None, expire=86400):
```

Comment:
missing newline

Reply:
:+1:

Reply:
actually it looks like this method is not used at all. Could you check?

Reply:
Yeah, you are right. It's leftover from processing additional files I did at first.